apiserver/params/network.go

```diff
@@ -1226,3 +1236,111 @@ type SubnetsResult struct {
 type SubnetsResults struct {
 	Results []SubnetsResult `json:"results"`
 }
+
+// SpaceInfosParams provides the arguments for a SpaceInfos call.
+type SpaceInfosParams struct {
+	// A list of space IDs for filtering the returned set of results. If
+	// empty, all spaces will be returned.
+	FilterBySpaceIDs []string `json:"space-ids,omitempty"`
+}
+
+// SpaceInfos represents the result of a SpaceInfos API call.
+type SpaceInfos struct {
+	Infos []SpaceInfo `json:"space-infos,omitempty"`
+}
+
+// SpaceInfo describes a space and its subnets.
+type SpaceInfo struct {
+	ID         string     `json:"id"`
+	Name       string     `json:"name"`
+	ProviderID string     `json:"provider-id,omitempty"`
+	Subnets    []SubnetV3 `json:"subnets,omitempty"`
+}
+
+// FromFromNetworkSpaceInfos converts a network.SpaceInfos into a serializable
+// payload that can be sent over the wire.
+func FromNetworkSpaceInfos(allInfos network.SpaceInfos) SpaceInfos {
```

Comment:
We've previously defined these in the relevant apiserver facade package but I think here is better actually.

Reply:
As mentioned in the extended commit log for this change, I moved them there because we _may_ need to implement `SpaceInfos` in other facades in the future (so we can consistently use SpaceInfos across the codebase) and having the marshal/unmarshal logic in `params` makes it very easy to clone the API method (also avoids the `networkingcommon` package pattern which I believe we are trying to get rid of eventually)